tests/framework/validators/FileValidatorTest.php

```diff
@@ -496,6 +522,7 @@ public function validMimeTypes()
             ['test.txt', 'text/*', 'txt'],
             ['test.xml', '*/xml', 'xml'],
             ['test.odt', 'application/vnd*', 'odt'],
+            ['test.tar.xz', 'application/x-xz', 'tar.xz'],
```

Comment:
```suggestion
            ['test.tar.xz', 'application/x-xz', 'xz'],
```

Reply:
Mime types are specified for topmost extension.

Reply:
In FileValidator.php:validateExtension in row 401 get extension from real filename.
After - checks mimetype from uploaded file, get all variants of extensions for this mimetype and compare this variants with real extension of file.
For same filename.tar.xz - real extension from row 401 - xz